app/src/main/java/com/kickstarter/ui/fragments/RewardsFragment.kt

```diff
@@ -0,0 +1,80 @@
+package com.kickstarter.ui.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.PagerSnapHelper
+import com.kickstarter.R
+import com.kickstarter.libs.BaseFragment
+import com.kickstarter.libs.qualifiers.RequiresFragmentViewModel
+import com.kickstarter.libs.utils.RewardDecoration
+import com.kickstarter.models.Project
+import com.kickstarter.ui.adapters.HorizontalRewardsAdapter
+import com.kickstarter.viewmodels.RewardFragmentViewModel
+import kotlinx.android.synthetic.main.fragment_rewards.*
+
+@RequiresFragmentViewModel(RewardFragmentViewModel.ViewModel::class)
+class RewardsFragment : BaseFragment<RewardFragmentViewModel.ViewModel>() {
+
+    private lateinit var project: Project
+
+    companion object {
+        private var PROJECT = "project"
+        fun newInstance(project: Project): RewardsFragment {
+            val args = Bundle()
+            args.putParcelable(PROJECT, project)
+            val fragment = RewardsFragment()
+            fragment.arguments = args
+            return fragment
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val args = arguments
+        if (args != null) {
+            project = args.getParcelable(PROJECT) as Project
```

Comment:
this model should be instantiated in the viewmodel using the `arguments` observable

Reply:
We were taking a look at how we implement this in other places and we couldn't find a place where we use arguments observable to instantiate a model from the viewModel. Could you point us some place where we use the arguments pattern, so we have a better understanding of how to do it?

Reply:
Yeah I actually do it in my branch here: https://github.com/kickstarter/android-oss/blob/io/pledge-fragment/app/src/main/java/com/kickstarter/viewmodels/PledgeFragmentViewModel.kt#L93

Reply:
Awesome. Thanks!